File: Backend/friendnest/src/main/java/pl/polsl/friendnest/service/UserService.java
package pl.polsl.friendnest.service;

import pl.polsl.friendnest.model.User;
import pl.polsl.friendnest.model.request.ChangeUserCredentialsRequest;

public interface UserService {
    User getUserById(Long userId);

    User changeUserCredentails(ChangeUserCredentialsRequest changeUserCredentialsRequest);
}

File: Frontend/friend-nest/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { ChangeUserCredentialsRequest } from '../models/request/ChangeUserCredentialsRequest';
import { Observable } from 'rxjs';
import { User } from '../models/User';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  constructor(private http: HttpClient) {}

  changeUserCredentails(
    changeUserCredentialsRequest: ChangeUserCredentialsRequest,
  ): Observable<User> {
    return this.http.post<User>(
      'http://localhost:8080/api/changeUserCredentails',
      changeUserCredentialsRequest,
    );
  }
}
